scripts/util/build-doc.js

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env node
+
+var path = require('path');
+var fs = require('fs');
+var yuidoc = require('./yuidoc');
+var NodeGit = require('nodegit');
+var util = require('util');
+var recursiveRmdir = require('./recursive-rmdir');
+var recursiveCpdir = require('./recursive-cpdir');
+
+function buildTreePromise(commitPromise, name) {
+  return commitPromise.then(function (commit){
+    return commit.getTree();
+  }).then(function (tree){
```

Comment:
sticking the `then` here couples it to the body of the previous promise. Since this is scheduled for after `commit.getTree()`, maybe nest it there?

```javascript
commit.getTree().then(function (tree) { ... })